### PR TITLE
feat: plug in query data into token details & add no info state

### PR DESCRIPTION
**Pull request by kayleegeorge**
plug in query data into token details, decompose token details skeleton (loading state, no info state, loaded details state)

### CODE REVIEW

File: src/components/Explore/TokenDetails/TokenDetail.tsx [hunk 1]
```diff
@@ -152,8 +156,111 @@ const NetworkBadge = styled.div<{ networkColor?: string; backgroundColor?: strin
...
+  font-size: 12px;
+  gap: 4px;
+  align-items: center;
+  border-bottom: 1px solid ${({ theme }) => theme.backgroundOutline};
+  padding: 8px 0px;
+  margin-top: -40px;
+`
+
+// todo: change duration
+const tokenDetailsStatsQuery = gql`
```
Comment (kayleegeorge) [inline src/components/Explore/TokenDetails/TokenDetail.tsx:180]: query @cartcrom

File: src/components/Explore/TokenDetails/TokenDetail.tsx [hunk 2]
```diff
@@ -5,14 +5,18 @@ import CurrencyLogo from 'components/CurrencyLogo'
...
 import { useAtomValue } from 'jotai/utils'
 import { darken } from 'polished'
-import { useCallback } from 'react'
+import { ReactElement, useCallback } from 'react'
 import { useState } from 'react'
-import { ArrowLeft, Copy, Heart } from 'react-feather'
+import { ArrowLeft, Copy, Heart, TrendingUp } from 'react-feather'
+import { usePreloadedQuery } from 'react-relay'
 import { Link, useNavigate } from 'react-router-dom'
+import { CHAIN_SUBGRAPH_URL } from 'state/data/slice'
```
Comment (cartcrom) [inline src/components/Explore/TokenDetails/TokenDetail.tsx:19]: The current graphql URLs in redux are for querying a separate set of data specific to Pools, it has no use here. Relay hook calls with automatically use the correct url once that's setup, so no need to worry about urls here

File: src/components/Explore/TokenDetails/TokenDetail.tsx [hunk 3]
```diff
@@ -171,28 +278,100 @@ export default function LoadedTokenDetail({ address }: { address: string }) {
   const chainInfo = getChainInfo(token?.chainId)
   const networkLabel = chainInfo?.label
   const networkBadgebackgroundColor = chainInfo?.backgroundColor
+  const truncatedTokenAddress = `${address.slice(0, 4)}...${address.slice(-3)}`
+  const tokenDetailsData = usePreloadedQuery(tokenDetailsStatsQuery, {
+    contract: {
+      address,
+      chain: token ? CHAIN_SUBGRAPH_URL[token.chainId] : CHAIN_SUBGRAPH_URL[SupportedChainId.MAINNET],
```
Comment (cartcrom) [inline src/components/Explore/TokenDetails/TokenDetail.tsx:285]: I believe chain here should just be the the token's chain id, not a url

File: src/state/data/slice.ts [hunk 4]
```diff
@@ -6,7 +6,7 @@ import { ClientError, gql, GraphQLClient } from 'graphql-request'
 import { AppState } from 'state'
 
 // List of supported subgraphs. Note that the app currently only support one active subgraph at a time
-const CHAIN_SUBGRAPH_URL: Record<number, string> = {
+export const CHAIN_SUBGRAPH_URL: Record<number, string> = {
```
Comment (cartcrom) [inline src/state/data/slice.ts:9]: undo all changes in this file since this url isn't needed elsewhere

**Review by cartcrom — CHANGES_REQUESTED** [review]

File: src/components/Tokens/TokenDetails/TokenDetail.tsx [hunk 5]
```diff
@@ -161,27 +268,95 @@ export default function LoadedTokenDetail({ address }: { address: string }) {
   const chainInfo = getChainInfo(token?.chainId)
   const networkLabel = chainInfo?.label
   const networkBadgebackgroundColor = chainInfo?.backgroundColor
+  const environment = useRelayEnvironment()
+  const queryReference = loadQuery(environment, tokenDetailsStatsQuery, {
+    contract: {
+      address,
+      chain: token ? token.chainId : SupportedChainId.MAINNET,
```
Comment (cmcewen) [inline src/components/Tokens/TokenDetails/TokenDetail.tsx:275]: i'm not sure falling back to mainnet is right here - imo the address + chainId go together as a unit, and in the worst case we would show the wrong token on a different network

File: src/components/Tokens/TokenDetails/TokenDetail.tsx [hunk 6]
```diff
@@ -161,27 +268,95 @@ export default function LoadedTokenDetail({ address }: { address: string }) {
...
+        breadcrumb={
+          <>
+            <ArrowLeft size={14} /> Explore
+          </>
+        }
+        tokenInfo={
+          <>
+            <TokenNameCell>
+              <CurrencyLogo currency={currency} size={'32px'} />
+              {!token ? 'No token name available' : token.name} <TokenSymbol>{token && token.symbol}</TokenSymbol>
```
Comment (cmcewen) [inline src/components/Tokens/TokenDetails/TokenDetail.tsx:293]: i18n this?

**Comment by kayleegeorge** [comment]
closing this for: https://github.com/Uniswap/interface/pull/4407 and will integrate query in another PR

**Pull request closed without merging**